internal/sys/sys_test.go

```diff
@@ -22,6 +22,8 @@ func TestContext_WalltimeNanos(t *testing.T) {
 }
 
 func TestDefaultSysContext(t *testing.T) {
+	t.Skip("fixme: temporarily skip")
```

Comment:
I can't really figure out why the two lists of files are no longer equal. I suspect the changes to `stdioFileReader` have to do with this, but I can't tell what's wrong. Maybe something related to func references?

Reply:
my spider sense was tingling the right way, [`func` types are not comparable](https://go.dev/ref/spec#Comparison_operators)

Reply:
strategy pattern it is... don't hit me I am a Java dev recovering 🤣 

```
type StdioFilePoller interface {
	Poll(duration time.Duration) (bool, error)
}

type pollerDefaultStdin struct{}
...

// Poll implements StdioFilePoller for pollerDefaultStdin
func (*pollerDefaultStdin) Poll(duration time.Duration) (bool, error) { ... }
var PollerDefaultStdin = &pollerDefaultStdin{}
```